enterprise/tpa_pipeline.py

```diff
@@ -144,6 +155,38 @@ def get_user_social_auth(user, enterprise_customer):
     return user_social_auth
 
 
+def get_default_idp_user_social_auth(enterprise_customer, user=None, user_idp_id=None):
+    """
+    Return social auth entry of user for given enterprise default IDP.
+
+    Arguments:
+        user (User): user object
+        enterprise_customer (EnterpriseCustomer): Instance of the enterprise customer.
+        user_idp_id (str): User id of user in third party LMS
+
+    """
+    default_provider = enterprise_customer.default_provider_idp
+
+    if default_provider:
+        tpa_provider = get_identity_provider(default_provider.provider_id)
+        filter_kwargs = {
+            'provider': tpa_provider.backend_name,
+            'uid__contains': tpa_provider.provider_id[5:]
```

Comment:
Can we avoid this string manipulation?

Reply:
I think No. Either we need to annotate `saml` with UID or remove it from provider id to match the UID.